app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.kt

```diff
@@ -30,7 +35,22 @@ class CalculatorModel : CalculatorContract.Model {
     }
 
     override fun onPressedOperation(operation: String) {
-        entity.setOperator(operation)
+        if (operation == MINUS) {
```

Comment:
Is there any way to simplify this flow ? theres too much nested ifs. Any ideas?

Reply:
I try to change the code by placing when declaration instead of if statement but the code stays same and sometimes even more complex.